modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCAggregateFunctionOnlineTest.java

```diff
@@ -660,4 +662,104 @@ public void testUniqueCountWithLimit() throws Exception {
         assertEquals(0, v.getUnique().size());
         assertEquals(2, result);
     }
+
+    @SuppressWarnings("unchecked")
+    public void testUniqueMultipleAttributes() throws Exception {
+        UniqueVisitor v = new UniqueVisitor("stringProperty", "doubleProperty");
+        dataStore.getFeatureSource(tname("ft4")).accepts(Query.ALL, v, null);
+        Set result = v.getResult().toSet();
+        assertEquals(6, result.size());
+        List<String> names = v.getAttrNames();
+        assertEquals("stringProperty", names.get(0));
+        assertEquals("doubleProperty", names.get(1));
+
+        LinkedList<Object> shouldContainOne = new LinkedList<>();
```

Comment:
I'd go for full expected results tests in this test class too.

Reply:
fixed